transport/awssqs/consumer.go

```diff
@@ -0,0 +1,296 @@
+package awssqs
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
+	"github.com/go-kit/kit/endpoint"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/transport"
+)
+
+// Delete is a type to indicate when the consumed message should be deleted
+type Delete int
+
+const (
+	// BeforeHandle deletes the message before starting to handle it.
+	BeforeHandle Delete = iota
+	// AfterHandle deletes the message once it has been fully processed.
+	// This is the consumer's default value.
+	AfterHandle
+	// Never does not delete the message.
+	Never
+)
+
+// Consumer wraps an endpoint and provides a handler for sqs messages.
+type Consumer struct {
```

Comment:
All pubsub implementations in this repository have their name `Subscriber`. Also on AWS doc said it's a pub/sub messaging, not pub/consumer.

Reply:
I found the term consumer more adapted given the fact that you need to send a receive request to retrieve messages from the queue.
Do you have quote and a link to prove your point ?

In their [pub/sub documentation](https://aws.amazon.com/pub-sub-messaging/) they actually oppose pub/sub systems to messaging queues, while in their [messaging queue documentation](https://aws.amazon.com/message-queue/) they use the terms producer/consumer. This would actually mean that Publisher should be replaced by Producer.

If this is ok for you @xyluet, I'll change Publisher by Producer.

Reply:
I'm actually not a maintainer. This will be the first one that using terms producer/consumer in this repo. What do you think @peterbourgon ?

Reply:
Let's use whatever the SDK prefers. No preference on Go kit side.

Reply:
Producer/consumer seems ok.

Reply:
Done